pixeltable/share/packager.py

```diff
@@ -69,138 +70,122 @@ def __init__(self, table: catalog.Table, additional_md: Optional[dict[str, Any]]
-        # these will be specially handled later)
-        select_exprs: dict[str, exprs.Expr] = {}
-
-        # As we generate the select list, we construct a separate list of column types. We can't rely on df._schema
-        # to get the column types, since we'll be substituting `fileurl`s for media columns.
-        actual_col_types: list[ts.ColumnType] = []
-
-        for col_name, col in t._tbl_version.get().cols_by_name.items():
-            if not col.is_stored:
-                continue
-            if col.col_type.is_media_type():
-                select_exprs[col_name] = t[col_name].fileurl
-            else:
-                select_exprs[col_name] = t[col_name]
-            actual_col_types.append(col.col_type)
-            if col.records_errors:
-                select_exprs[f'{col_name}_errortype'] = t[col_name].errortype
-                actual_col_types.append(ts.StringType())
-                select_exprs[f'{col_name}_errormsg'] = t[col_name].errormsg
-                actual_col_types.append(ts.StringType())
-
-        # Run the select() on `self.table`, not `t`, so that we export only those rows that are actually present in
-        # `self.table`.
-        df = self.table.select(**select_exprs)
-        namespace = self.__iceberg_namespace(t)
-        self.iceberg_catalog.create_namespace_if_not_exists(namespace)
-        iceberg_schema = self.__to_iceberg_schema(df._schema)
-        iceberg_tbl = self.iceberg_catalog.create_table(f'{namespace}.{t._name}', schema=iceberg_schema)
-
-        # Populate the Iceberg table with data.
-        # The data is first loaded from the DataFrame into a sequence of pyarrow tables, batched in order to avoid
-        # excessive memory usage. The pyarrow tables are then amalgamated into the (single) Iceberg table on disk.
-        for pa_table in self.__to_pa_tables(df, actual_col_types, iceberg_schema):
-            iceberg_tbl.append(pa_table)
+        sql_types = {col.name: col.type for col in tv.store_tbl.sa_tbl.columns}
+        media_cols: set[str] = set()
+        for col in tv.cols_by_name.values():
+            if col.is_stored and col.col_type.is_media_type():
+                media_cols.add(col.store_name())
 
-    @classmethod
-    def __iceberg_namespace(cls, table: catalog.Table) -> str:
-        """
-        Iceberg tables must have a namespace, which cannot be the empty string, so we prepend `pxt` to the table path.
-        """
-        parent_path = table._parent()._path()
-        if len(parent_path) == 0:
-            return 'pxt'
-        else:
-            return f'pxt.{parent_path}'
+        parquet_schema = self.__to_parquet_schema(tv.store_tbl.sa_tbl)
+        # The parquet file naming scheme anticipates future support for partitioning.
+        parquet_dir = self.tables_dir / f'tbl_{tv.id.hex}'
+        parquet_dir.mkdir()
+        parquet_file = parquet_dir / f'tbl_{tv.id.hex}.00000.parquet'
```

Comment:
Is there a single file per table?

Might be good to break it up into smaller pieces and then do parallel upload. Leave todo?

Reply:
There's a comment in there already about future partitioning? I'll make it a TODO. For parallel uploads, we'd have to chunk the tarball, right?